pkg/v1/layout/write.go

```diff
@@ -264,6 +264,15 @@ func (l Path) writeLayer(layer v1.Layer) error {
 	return l.WriteBlob(d, r)
 }
 
+// RemoveBlob removes a file from the blobs directory in the Path
+// at blobs/{hash.Algorithm}/{hash.Hex}
+// It does *not* remove any reference to it from other manifests or indexes, or
+// from the root index.json.
+func (l Path) RemoveBlob(hash v1.Hash) error {
+	dir := l.path("blobs", hash.Algorithm)
+	return os.RemoveAll(filepath.Join(dir, hash.Hex))
```

Comment:
I think we just want `Remove` not `RemoveAll`?

Reply:
Because I don't actually want the action of removing it; I want it not to be there. `RemoveAll` handles it not being there, while `Remove` errors if the file doesn't exist. That is easy enough to handle with `os.IsNotExist`; I was just lazy.

I can fix it.

Reply:
OK, I changed this part.

Reply:
Ah, that wasn't super clear at a glance. Explicitly checking for not found is easier for me to understand 👍